Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,514 @@
+  -- decode Memℒp into a.e. strong measurability and finite snorm
+  obtain ⟨haesmf, hsnf⟩ := hf
+  -- transform snorm to lintegral
+  rw [snorm_eq_snorm' (by assumption) (by assumption)] at hsnf
+  have hinpf := calc
+    ∫⁻ a, ‖f a‖₊ ^ p.toReal ∂μ
+      = (snorm' f p.toReal μ) ^ p.toReal := lintegral_rpow_nnnorm_eq_rpow_snorm' hrp_pos
+    _ < ∞                                := (rpow_lt_top_iff_of_pos hrp_pos).mpr hsnf
+  -- get a.e. measurability for the integrand
+  -- XXX: Why does `AEStronglyMeasurable.ennnorm` only give the weaker AEMeasurable?
+  --      It would make sense to me to use `haesmf.ennnorm.aemeasurable` below.
+  have haemnf := haesmf.ennnorm
+  have haemnpf := haemnf.pow_const p.toReal
+  -- use core result for lintegral (needs only AEMeasurable), the target estimate will be in `hsfε`
+  obtain ⟨s, hms, hμs, hsfε⟩ := lintegral_indicator_compl_le haemnpf hinpf hεp
+  use s, hms, hμs
+  -- move indicator through function compositions, XXX: is this simp-able?
+  rw [← Function.comp_def (fun x : ℝ≥0∞ => x ^ p.toReal)] at hsfε
+  rw [← Function.comp_def ENNReal.ofNNReal] at hsfε
+  rw [← Function.comp_def nnnorm] at hsfε
+  rw [sᶜ.indicator_comp_of_zero (@ENNReal.zero_rpow_of_pos p.toReal hrp_pos)] at hsfε
+  rw [sᶜ.indicator_comp_of_zero ENNReal.coe_zero] at hsfε
+  rw [sᶜ.indicator_comp_of_zero nnnorm_zero] at hsfε
+  rw [Function.comp_def nnnorm] at hsfε
+  rw [Function.comp_def ENNReal.ofNNReal] at hsfε
+  rw [Function.comp_def (fun x : ℝ≥0∞ => x ^ p.toReal)] at hsfε
+  -- convert lintegral to snorm
+  rw [lintegral_rpow_nnnorm_eq_rpow_snorm' hrp_pos] at hsfε
+  rw [← snorm_eq_snorm' (by assumption) (by assumption)] at hsfε
+  -- commute ENNReal coersion with rpow, use rpow monotonicity
+  rw [← ofReal_rpow_of_pos (by assumption)] at hsfε
+  rw [ENNReal.rpow_le_rpow_iff hrp_pos] at hsfε
+  exact hsfε
+
+/-- A constant function is tight. -/
+theorem unifTight_const {g : α → β} (hp : 1 ≤ p) (hp_ne_top : p ≠ ∞) (hg : Memℒp g p μ) :
+    UnifTight (fun _ : ι => g) p μ := fun ε hε ↦ by
+  by_cases hε_top : ε = ∞
+  · exact ⟨∅, (by measurability), fun _ => hε_top.symm ▸ le_top⟩
+  have hrε : 0 < ε.toReal := ENNReal.toReal_pos hε.ne' hε_top
+  obtain ⟨s, _, hμs, hgε⟩ := hg.snorm_indicator_compl_le hp hp_ne_top hrε
+  exact ⟨s, ne_of_lt hμs, fun _ => hgε.trans_eq (ENNReal.ofReal_toReal hε_top)⟩
+
+/-- A single function is tight. -/
+theorem unifTight_subsingleton [Subsingleton ι] (hp_one : 1 ≤ p) (hp_top : p ≠ ∞)
+    {f : ι → α → β} (hf : ∀ i, Memℒp (f i) p μ) : UnifTight f p μ := fun ε hε ↦ by
+  by_cases hε_top : ε = ∞
+  · exact ⟨∅, (by measurability), fun _ => hε_top.symm ▸ le_top⟩
+  have hrε : 0 < ε.toReal := ENNReal.toReal_pos hε.ne' hε_top
+  by_cases hι : Nonempty ι
+  case neg => exact ⟨∅, (by measurability), fun i => False.elim <| hι <| Nonempty.intro i⟩
+  cases' hι with i
+  obtain ⟨s, _, hμs, hfε⟩ := (hf i).snorm_indicator_compl_le hp_one hp_top hrε
+  refine' ⟨s, ne_of_lt hμs, fun j => _⟩
```

Comment:
Please use `refine` instead of `refine'` (everywhere in the PR).
```suggestion
  refine ⟨s, ne_of_lt hμs, fun j => ?_⟩
```

Reply:
Done in next update.